tests/queryset/queryset.py

```diff
@@ -1763,6 +1764,27 @@ class BlogPost(AbstractBlogPost):
         self.assertEqual(post.authors, [me])
         self.assertEqual(another.authors, [])
 
+    def test_update_with_limits(self):
+        """Ensure that the update regards limit.
+        """
+
+        class Item(Document):
+            value = IntField()
+
+        Item.drop_collection()
+
+        for i in range(1, 6):
+            Item(value=i).save()
+
+        Item.objects[:3].update(set__value=88)
```

Comment:
Set all equal values in the DB like so. [2, 2, 2, 2, 2].

Then do:

                Item.objects(value=2)[:2].update(value=3)
                result = Item.objects(value=2)
                self.assertEqual(result, [2, 2, 2])


The other test would be the same as above. Set all items to 2 in the db.

                 Item.object(value=2).limit(2).update(value=3)
                 result = Item.objects(value=2)
                 self.assertEqual(result, [2, 2, 2])